test/backend/test_dynamic_shape.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+import tensorflow as tf
+import unittest
+
+from onnx_tf.backend import onnx_graph_to_tensorflow_rep
+from onnx_tf.common.legacy import legacy_opset_pre_ver
+from onnx import helper
+from onnx import TensorProto
+
+
+# Run the following test in graph mode
+tf.compat.v1.disable_eager_execution()
+
+class TestDynamicShape(unittest.TestCase):
```

Comment:
Not yet familiar with this codebase too much, but what I was thinking in https://github.com/onnx/onnx-tensorflow/issues/546 is that we can abstract the `TensorflowBackend.run_node` in the `test_nodes.py` so that it would construct dynamic shape test cases out of the current static ones automatically. Something like:

```python
# test_nodes.py

def run_node(..., test_dynamic_shape=True):
    TensorflowBackend.run_node(...) # test the static shape
    if test_dynamic_shape:
        # create a graph with placeholder inputs for all tensor arguments
        # feed the arguments to the graph
```
Do you think that works?

Reply:
Agreed having some generic code to test dynamic shapes is a good idea. The logic to test dynamic shapes is slightly different, having to construct the graph with dummy tensors and run the graph with real inputs. Since this is the first time we support dynamic shapes with unit tests, I think we could leave the current test code as-is and work on a PR later to generalize it. Of course I don't have a problem if @winnietsang is going to include the generic handling in this PR.

Reply:
I agree this is a great idea, it can be a future enhancement to our node unit test. But I want to focus on verify our handlers can handle dynamic shape first. Having a separate test file for dynamic shape testing, is a simpler approach.